packages/devtools_app/lib/src/screens/memory/panes/allocation_tracing/allocation_profile_tracing_tree.dart

```diff
@@ -0,0 +1,274 @@
+
+import '../../../../primitives/utils.dart';
+import '../../../../shared/common_widgets.dart';
+import '../../../../shared/table.dart';
+import '../../../../shared/table_data.dart';
+import '../../../../shared/theme.dart';
+import '../../../../shared/utils.dart';
+import '../../../profiler/cpu_profile_columns.dart';
+import '../../../profiler/cpu_profile_model.dart';
+import 'allocation_profile_tracing_view_controller.dart';
+
+const double _countColumnWidth = 130;
+
+/// Displays an allocation profile as a tree of stack frames, displaying
+/// inclusive and exclusive allocation counts.
+class AllocationTracingTree extends StatelessWidget {
+  const AllocationTracingTree({required this.controller});
+
+  final AllocationProfileTracingViewController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder<TracedClass?>(
+      valueListenable: controller.selectedTracedClass,
+      builder: (context, selection, _) {
+        Widget? errorColumn;
+        if (selection == null) {
+          errorColumn = _allocationTracingInstructions(context);
+        } else if (!selection.traceAllocations) {
+          return Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              Text(
+                'Allocation tracing is not enabled for class ${selection.cls.name}.\n',
+              ),
+              _allocationTracingInstructions(context),
+            ],
+          );
+        } else if (selection.traceAllocations &&
+            controller.selectedTracedClassAllocationData == null) {
+          errorColumn = Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              Text(
+                'No allocation samples have been collected for class ${selection.cls.name}.\n',
+              ),
+            ],
+          );
+        }
+        if (errorColumn != null) {
+          return errorColumn;
+        }
+        late Function(Function()) updateTreeStateCallback;
+        final theme = Theme.of(context);
+        return Column(
+          children: [
+            AreaPaneHeader(
+              title: Text.rich(
+                TextSpan(
+                  children: [
+                    const TextSpan(
+                      text: 'Traced allocations for: ',
+                    ),
+                    TextSpan(
+                      style: theme.fixedFontStyle,
+                      text: controller.selectedTracedClass.value?.cls.name!,
+                    ),
+                  ],
+                ),
+              ),
+              tall: true,
+              needsTopBorder: false,
+              actions: [
+                ExpandAllButton(
+                  onPressed: () => updateTreeStateCallback(
+                    () {
+                      final data = controller.selectedTracedClassAllocationData;
+                      if (data == null) {
+                        return;
+                      }
```

Comment:
if `data == null` we should disable these buttons instead of having a pressed function that does nothing. Setting the `onPressed` param to null should disable the button

Reply:
Actually, this case should never happen since we won't be displaying this table if `selectedTracedClass == null`, so we should just null assert on `data`.